mkosi/__init__.py

```diff
@@ -1996,12 +1982,12 @@ def invoke_dnf(
     # On Debian, rpm/dnf ship with a patch to store the rpmdb under ~/
     # so it needs to be copied back in the right location, otherwise
     # the rpmdb will be broken. See: https://bugs.debian.org/1004863
-    rpmdb_home = root / "root/.rpmdb"
+    rpmdb_home = state.root / "state.root/.rpmdb"
```

Comment:
Maybe a case of too much search and replace?
```suggestion
    rpmdb_home = state.root / "root/.rpmdb"
```